config/balanceworker.go

```diff
@@ -1,19 +1,25 @@
 package config
 
 import (
+	"errors"
+
 	"github.com/spf13/viper"
+
+	"github.com/openmeterio/openmeter/pkg/errorsx"
 )
 
 type BalanceWorkerConfiguration struct {
 	ConsumerConfiguration `mapstructure:",squash"`
 }
 
 func (c BalanceWorkerConfiguration) Validate() error {
+	var errs []error
+
 	if err := c.ConsumerConfiguration.Validate(); err != nil {
-		return err
+		errs = append(errs, errorsx.WithPrefix(err, "consumer"))
```

Comment:
Nit: I think it would be better if the prefixes used the full name of the configuration object, it would make things 100% unambiguous (e.g. consumer => ConsumerConfiguration)
Wdyt?

Reply:
Hm...maybe

Right now it follows the name as it appears in the config file which probably makes it easier to debug a problem.

Ultimately, this should be a "path" to the configuration (whether it's file or not), but people rarely look at the code to figure out configuration.